pandas/core/strings.py

```diff
@@ -150,14 +150,19 @@ def _map_stringarray(
         an ndarray.
 
     """
-    from pandas.arrays import IntegerArray, StringArray
+    from pandas.arrays import IntegerArray, StringArray, BooleanArray
 
     mask = isna(arr)
 
     assert isinstance(arr, StringArray)
     arr = np.asarray(arr)
 
-    if is_integer_dtype(dtype):
+    if is_integer_dtype(dtype) or is_bool_dtype(dtype):
+        if is_integer_dtype(dtype):
```

Comment:
MyPy apparently doesn't like this...

```
pandas/core/strings.py:164: error: Incompatible types in assignment (expression has type "Type[BooleanArray]", variable has type "Type[IntegerArray]")
```

Any suggestions on how to please the type checker?

Reply:
Yea if you have assignment in an `if...else` block the type is inferred from the first one that appears.

So before the block you can just declare `constructor: Type[Union[IntegerArray, BooleanArray]]` or maybe even something simpler like `constructor: Type[ExtensionArray]` depending on what is valid

Reply:
nvm, I think I got it.